unstructured/partition/pdf.py

```diff
@@ -92,11 +92,11 @@ def partition_pdf(
 
 
 def partition_pdf_or_image(
-    filename: str = "",
+    filename: Optional[str] = "",
     file: Optional[Union[bytes, BinaryIO, SpooledTemporaryFile]] = None,
     is_image: bool = False,
     include_page_breaks: bool = False,
-    strategy: str = "auto",
+    strategy: Optional[str] = None,
```

Comment:
Let's keep the default "auto", and the logic should route to "hi_res" if "auto" is selected and GPUs are available.